plugins/doc_fragments/template.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) IBM Corporation 2022, 2023
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+class ModuleDocFragment(object):
+
+    DOCUMENTATION = r'''
+options:
+  use_template:
+    description:
+      - Whether the module should treat C(src) as a Jinja2 template and
+        render it before continuing with the rest of the module.
+      - Only valid when C(src) is a local file or directory.
+      - All variables defined in inventory files, vars files and the playbook
```

Comment:
Do you know what is the order of precedence when getting the variables from different source? Might be worth it adding into the docs.

Reply:
The[ Ansible documentation has the complete hierarchy](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#variable-precedence-where-should-i-put-a-variable) the engine uses. I'll add a note about it and link it in the doc fragment.